TravelAgencyService/TravelAgencyService/Services/AgencyOperationsService.cs

```diff
@@ -0,0 +1,57 @@
+using TravelAgencyService.Models;
+
+namespace TravelAgencyService.Services;
+
+public class AgencyOperationsService
+{
+    private readonly ConsoleService _consoleService;
+
+    public AgencyOperationsService(ConsoleService consoleService)
+    {
+        _consoleService = consoleService;
+    }
+    public void BookTicket(AgencyModel travelAgency)
+    {
+        const int maxRetryAttempts = 3;
+        const int delayMilliseconds = 1000;
+        var retryAttempts = 0;
+
+        while (retryAttempts < maxRetryAttempts)
+        {
+            var destination = _consoleService.GetDestination();
+            try
+            {
+                if (destination != null) travelAgency.BookTicket(destination);
+                _consoleService.ShowMessage("Booking is successful.");
+                break;
+            }
+            catch (Exception ex)
+            {
+                _consoleService.ShowMessage($"Attempt {retryAttempts + 1} failed: {ex.Message}");
+                retryAttempts++;
+                if (retryAttempts < maxRetryAttempts)
+                {
+                    Thread.Sleep(delayMilliseconds);
```

Comment:
You can also use async analogue Task.Delay but lets keep as it is as you haven't learn async/await yet